### ISSUE TITLE
Add the ability to optionally filter sam log output

### ISSUE THREAD

**Issue by awsjeffg** [aws/aws-sam-cli#2188]
as per Ben:
https://twitter.com/ben11kehoe/status/1298959222831366144?s=20

"It annoys me that the Lambda logs policy in SAM still allows access to all log groups, rather than introducing a v2 that would scope it narrowly to just that Lambda’s group "

While we should not change the current behavior of sam log, we could add a command-line option to filter output now and recommend customers add this option to their samconfig.toml. When we design a sam cli v2 we can revisit how sam logs should work by default.

**Comment by awsjeffg** [comment]
@benkehoe

**Comment by benkehoe** [comment]
This was a comment about the IAM policy generated for an `AWS::Serverless::Function` resource to allow the function to write logs (and therefore part of the SAM transform), rather than the behavior of `sam log`. In particular, a better IAM policy would only provide permissions to the log group used by that function (note, this is the group that `sam log` fetches from). Currently, the policy allows the function to write to _any_ log group, which is extremely rarely needed.

I opened the issue as https://github.com/aws/serverless-application-model/issues/395, but it was closed as the existing log policy cannot be changed without breaking backwards compatibility, without exploring options for letting users opt in to a better-scoped policy.
